packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -1504,8 +1555,12 @@ export async function syncEngineApplyActiveTargetsChange(
   }
 
   for (const targetId of added) {
-    if (syncEngineImpl.queriesByTarget.has(targetId)) {
-      // A target might have been added in a previous attempt
+    // A target might have been added in a previous attempt, or could have listened to cache,
+    // but no remote store connection has been established by the primary client.
+    const targetHasBeenListenedTo =
```

Comment:
Maybe change it to `targetAlreadyListeningToRemoteStore`.

Also, the comment could be:

A target is already listening to remote store if it is already registered to `sharedClientState`.